status/metrics.go

```diff
@@ -18,108 +20,217 @@ const (
 )
 
 // Cardinality is limited to # objects * # conditions * # objectives
-var ConditionDuration = pmetrics.NewPrometheusHistogram(
-	metrics.Registry,
-	prometheus.HistogramOpts{
+var ConditionDuration pmetrics.ObservationMetric
+
+func conditionDurationMetric(objectName string, enableObjectBasedMetric bool) pmetrics.ObservationMetric {
```

Comment:
To me -- this seems kind of a weird way to handle creating the ConditionDuration metric -- this is only going to be assigned the first time this function is used -- which isn't really what you want for globals -- you'd want them to be instantiated as soon as the package is imported so you don't accidentally have folks importing and using `nil` values for things that they wouldn't expect to be `nil`